content/300-guides/500-other/880-troubleshooting-orm/100-help-articles/800-implicit-to-explicit-conversion.mdx

```diff
@@ -0,0 +1,146 @@
+
+```prisma
+model UserPost {
+  id        Int       @id @default(autoincrement())
+  userId    Int
+  postId    Int
+  user      User      @relation(fields: [userId], references: [id])
+  post      Post      @relation(fields: [postId], references: [id])
+  createdAt DateTime  @default(now())
+
+  @@unique([userId, postId])
+}
+```
+
+Once we create the join table, we would need to update our existing models that would now look like this:
+
+```prisma
+model User {
+  id        Int        @id @default(autoincrement())
+  name      String
+  posts     Post[]
+  userPosts UserPost[]
+}
+
+model Post {
+  id        Int        @id @default(autoincrement())
+  title     String
+  authors   User[]
+  userPosts UserPost[]
+}
+```
+
+If you are using Prisma Migrate, then you can invoke this command:
+
+```terminal
+npx prisma migrate dev --name "added explicit relation"
+```
+
+The migration will create the `UserPost` table and create one-to-many relation of `User` and `Post` model with `UserPost` model.
+
+**Migrating Existing data from implicit relation table to newly created join table**
+
+To migrate the existing data from the implicit relation table to the new explicit join table, you'll need to write a custom migration script. You can use the Prisma Client to interact with the database, read data from the implicit relation table, and write it to the new join table.
+
+Considering the above `User` and `Post` models, here’s an example script you can use to migrate data.
+
+```typescript
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+// A `main` function so that you can use async/await
+async function main() {
+  try {
+    // Fetch all users with their related posts
+    const users = await prisma.user.findMany({
+      include: { posts: true },
+    });
+
+    // Iterate over users and their posts, then insert records into the UserPost table
+    for (const user of users) {
+      for (const post of user.posts) {
+        await prisma.userPost.create({
+          data: {
+            userId: user.id,
+            postId: post.id,
+          },
+        });
+      }
+    }
+
+    console.log("Data migration completed.");
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+main()
+  .catch((e) => {
+    throw e;
+  })
+  .finally(async () => {
+    await prisma.$disconnect();
+  });
+```
+
+Once the data is migrated to the join Table, you can remove the implicit relation columns ( `posts` in `User` model and `author` in `Post` model ) as shown below:
+
+```prisma
+model User {
+  id        Int        @id @default(autoincrement())
+  name      String
+  userPosts UserPost[]
+}
+
+model Post {
+  id        Int        @id @default(autoincrement())
+  title     String
+  userPosts UserPost[]
+}
+```
```

Comment:
This could be a diff view that still includes the old relation fields, but marks them red as to be removed.

Reply:
Thanks for the suggestion! 🙌 
Added the diff view.